crates/pera-core/src/dwallet_mpc/mpc_session.rs

```diff
@@ -182,31 +289,39 @@ impl DWalletMPCSession {
     }
 
     /// Stores a message in the pending messages map.
-    /// The code stores every new message it receives for that session,
-    /// and when we reach the end of delivery,
-    /// we will advance the session if we have a threshold of messages.
-    fn store_message(&mut self, round: usize, message: &DWalletMPCMessage) -> DwalletMPCResult<()> {
-        let epoch_store = self.epoch_store()?;
-        let party_id = authority_name_to_party_id(&message.authority, &epoch_store)?;
-        if self.pending_messages[round].contains_key(&party_id) {
-            return Err(DwalletMPCError::MaliciousParties(vec![party_id]));
+    /// Every new message received for a session is stored.
+    /// When a threshold of messages is reached, the session advances.
+    fn store_message(&mut self, message: &DWalletMPCMessage) -> DwalletMPCResult<()> {
+        let source_party_id =
+            authority_name_to_party_id(&message.authority, &*self.epoch_store()?)?;
```

Comment:
Signature verification happens before this right? We know `message.authority` signed this message?

Reply:
Point us to where this occurs please